### ISSUE TITLE
[@typescript-eslint/ban-types] Can't use banned type name as class name in type declaration

### ISSUE THREAD

**Issue by SebastiaanYN** [typescript-eslint/typescript-eslint#830]
**Repro**

Given the following code:
```ts
/* eslint @typescript-eslint/ban-types: ['error'] */

export class Symbol {}

const symbol: Symbol = new Symbol();
```
ESLint gives the error:
```
5:15  error  Don't use 'Symbol' as a type. Use symbol instead  @typescript-eslint/ban-types
```

**Expected Result**
The code does not give an error, as `Symbol` is a different class than the one in JavaScript itself. Changing it to `symbol` makes the code incorrect. This also occurs with other banned types.

**Actual Result**
ESLint gives an error.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.13.0` |
| `@typescript-eslint/parser`        | `1.13.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `6.1.0` |
| `node`                             | `11.8.0` |
| `npm`                              | `6.10.3` |

**Comment by bradzacher** [comment]
How come you're shadowing the built in `Symbol` class?
Naming your code using the same names as builtins is generally discouraged because it can make it hard to debug problems eg due to failed imports.

---

The ban types rule isn't intended to be an advanced tool which determines imported vs global namespaces, etc. It just checks type names and suggests preferred alternatives.

The default config is intended to provide some sensible defaults such as not using the capitalised type names. You will have to reconfigure the rule if you want to shadow the global Symbol name within your codebase.
